server/register.go

```diff
@@ -2,22 +2,31 @@ package server
 
 import (
 	"errors"
+	"fmt"
 
 	"github.com/cashshuffle/cashshuffle/message"
 )
 
 // registerClient registers a new session.
 func (pi *packetInfo) registerClient() error {
+	var player *playerData
 	if len(pi.message.Packet) == 1 {
 		signed := pi.message.Packet[0]
 
 		if signed.GetSignature() == nil {
 			p := signed.GetPacket()
 			registration := p.GetRegistration()
 
-			if p.FromKey.String() != "" && registration != nil {
+			verificationKey := p.FromKey.String()
+			player = pi.tracker.playerByVerificationKey(verificationKey)
+			if player != nil {
+				return fmt.Errorf("server already has a player "+
+					"with verification key %s", verificationKey)
+			}
+
+			if verificationKey != "" && registration != nil {
 				player := playerData{
```

Comment:
NIT: shouldn't this just be `player =` since it is defined above?

Reply:
Yes definitely. Thank you. 0be99d3